main.cpp

```diff
@@ -18,16 +18,45 @@ using namespace clang::ast_matchers;
 using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
+private:
+    Rewriter& rewriter_;
 public:
-    CastCallBack(Rewriter& rewriter) {
-        // Your code goes here
-    };
+    CastCallBack(Rewriter& rewriter) : rewriter_(rewriter) { }
 
     void run(const MatchFinder::MatchResult &Result) override {
         // Your code goes here
+        const auto *Item = Result.Nodes.getNodeAs<CStyleCastExpr>("cast");
+        if (Item ->getCastKind() == CK_ToVoid)
+            return;
+        if (Item ->getExprLoc().isMacroID())
+            return;
+        SourceManager &SourceM = *Result.SourceManager;
+
+        auto ReRange = CharSourceRange::getCharRange (Item->getLParenLoc(),

```

Comment:
Please descrbe the difference between the use of `getCharRange()` and `getTokenRange()` with examples.

Reply:
There is no answer to the question.